### PR TITLE
Return errors from SSE requests to the client

### PR DESCRIPTION
**Pull request by jakemac53**
Fixes https://github.com/dart-lang/webdev/issues/663

When we have an error handling an SSE request we now return that to the client and log it there.

These are generally believed to be race condition related errors so the client in most cases would already be disconnected (and we swallow the error if the response sink is closed).

This should effectively hide errors related to the app disconnecting in the middle of handling a request, while still surfacing real errors in the browser console.

cc @jonahwilliams

### CODE REVIEW

File: dwds/lib/src/handlers/dev_handler.dart [hunk 1]
```diff
@@ -157,122 +158,137 @@ class DevHandler {
...
-        if (_devTools == null) {
-          injectedConnection.sink
-              .add(jsonEncode(serializers.serialize(DevToolsResponse((b) => b
-                ..success = false
-                ..error = 'Debugging is not enabled.\n\n'
-                    'If you are using webdev please pass the --debug flag.\n'
-                    'Otherwise check the docs for the tool you are using.'))));
-          return;
-        }
+      try {
```
Comment (grouma) [inline dwds/lib/src/handlers/dev_handler.dart]: [optional] This method is becoming very unruly imo. Worth breaking it up into smaller well named methods for clarity.

Comment (jakemac53) [reply dwds/lib/src/handlers/dev_handler.dart]: hide whitespace changes :D

but true, I can take a look at refactoring it

Comment (jakemac53) [reply dwds/lib/src/handlers/dev_handler.dart]: Ok done, PTAL

**Comment by jakemac53** [comment]
This is released now as `dwds` version `0.6.2`.

**Pull request merged**